uxarray/core/api.py

```diff
@@ -32,23 +29,14 @@ def open_grid(grid_filename_or_obj: Union[str, Path, xr.DataArray, np.ndarray,
         ``xr.DataArray``, ``np.ndarray``, ``list``, or ``tuple`` as a vertices
         object to define the grid.
 
-    islatlon : bool, optional
+    latlon : bool, optional
             Specify if the grid is lat/lon based
 
-    isconcave: bool, optional
-        Specify if this grid has concave elements (internal checks for this are possible)
-
-    gridspec: str, optional
-        Specifies gridspec
-
-    vertices: bool, optional
-        Whether to create grid from vertices
-
     source_grid: str, optional
         Path or URL to the source grid file. For diagnostic/reporting purposes only.
```

Comment:
Looks like `source_grid` is no longer used. The same "backwards-compatibility" in the previous comment applies to this one. Furthermore, it was discussed to be important for diagnostics purposes in the past.

Reply:
I'll introduce it back, I did not consider this when doing a sweep over the code.

Reply:
> Looks like source_grid is no longer used. The same "backwards-compatibility" issue in the previous comment applies to this one. Furthermore, it was discussed to be important for diagnostics purposes in the past.

I'll add a warning in `ux.open_grid` which explains it's removal.

Reply:
> Thanks, but the current deprecation warning doesn't help with this variable removal still leading to break the older workflows (even the most recent September version).
> 
> What I actually meant was that we shouldn't do such changes, even simple variable renamings, right away with the first PRs. Instead, we should follow such a flow, when considering changes in the user API:
> 
> 1. Create the issue that describes the changes that will happen in the user API (variable renaming, variable removal, function renaming, etc.)
> 2. Create a PR that does not perform the actual changes but only does start throwing DeprecationWarning for such user API, with an information of when the change will occur in the future
> 3. When the time comes to actual change, create the PR that performs the actual change and closes the issue.
> 
> That said, this PR should not do such actual user API changes. We should instead create an issue that describes all such user APi changes of this PR, and perform its tasks in the future.

I'll start doing this moving forward, this is extremely helpful! 

What would you suggest doing for this PR then?

Reply:
As we are agreed on that, we can get this PR go forward as is (with those changes), but in the future, we should better employ the above flow.

Reply:
> but in the future, we should better employ the above flow.

Will keep this in mind! Appreciate the advice.